app/scripts/views/sms_send.js

```diff
@@ -0,0 +1,164 @@
+    getAccount () {
+      // TODO - remove the `|| ...` when done with development
+      return this.model.get('account') || this.user.getSignedInAccount();
+    },
+
+    context () {
+      const escapedLearnMoreAttributes =
+          `id="learn-more" href="${encodeURI(View.LEARN_MORE_LINK)}" target="_learn-more" data-flow-event="link.learn_more"`;
+
+      return {
+        country: this.model.get('country'),
+        escapedLearnMoreAttributes
+      };
+    },
+
+    showChildView (ChildView, options = {}) {
+      // TODO remove the duplication between here and connect_another_device
+
+      // an extra element is needed to attach the child view to, the extra element
+      // is removed from the DOM when the view is destroyed. Without it, .child-view
+      // is removed from the DOM and a 2nd child view cannot be displayed.
+      this.$('.child-view').append('<div>');
+      options.el = this.$('.child-view > div');
+      const childView = this._createView(ChildView, options);
+      return childView.render()
+        .then(() => this.trackChildView(childView));
+    },
+
+    submit () {
+      const account = this.getAccount();
+      const phoneNumber = this._getPhoneNumber();
+
+      return account.sendSms(phoneNumber, SmsMessageIds.FIREFOX_MOBILE_INSTALL)
+        .then(() => this._onSendSmsSuccess(phoneNumber))
+        .fail((err) => this._onSendSmsError(err));
+    },
+
+    /**
+     * Get the user entered phone number, with country code
+     *
+     * @returns {String}
+     */
+    _getPhoneNumber () {
+      const phoneNumber = this.getElementValue(SELECTOR_PHONE_NUMBER);
+      const country = this.model.get('country');
+      const countryInfo = CountryTelephoneInfo[country];
+      if (phoneNumber.indexOf(countryInfo.prefix) === -1) {
+        return `${countryInfo.prefix}${phoneNumber}`;
+      }
+      return phoneNumber;
+    },
+
+    /**
+     * Sending an SMS to `phoneNumber` succeeded
+     *
+     * @param {String} phoneNumber
+     * @private
+     */
+    _onSendSmsSuccess (phoneNumber) {
+      this.navigate('sms/sent', {
+        country: this.model.get('country'),
+        phoneNumber
+      });
+    },
+
+    /**
+     * Sending an SMS failed with error `err`. Certain
+     * errors are handled inline, all others are re-thrown
+     * to be handled at a higher level.
+     *
+     * @param {Error} err
+     * @throws {Error}
+     * @private
+     */
+    _onSendSmsError (err) {
+      if (AuthErrors.is(err, 'INVALID_PHONE_NUMBER') || // auth-server validation
+          SmsErrors.is(err, 'INVALID_PHONE_NUMBER') || // nexmo validation
+          SmsErrors.is(err, 'UNROUTABLE_MESSAGE') ||
+          SmsErrors.is(err, 'NUMBER_BLOCKED')) {
+        this.showValidationError(this.$(SELECTOR_PHONE_NUMBER), err);
+        return;
+      }
+      // all other errors are handled at a higher level.
+      throw err;
+    }
+  }, {
+    LEARN_MORE_LINK: 'https://www.mozilla.org/en-US/privacy/websites/#campaigns'
```

Comment:
If this were used by more than one module (outside of tests), I'd say "you betcha", but the LEARN_MORE_LINK is only used here and in the corresponding tests, so I made it a static variable on the module.